copy/copy.go

```diff
@@ -668,7 +733,32 @@ type diffIDResult struct {
 
 // copyLayer copies a layer with srcInfo (with known Digest and possibly known Size) in src to dest, perhaps compressing it if canCompress,
 // and returns a complete blobInfo of the copied layer, and a value for LayerDiffIDs if diffIDIsNeeded
-func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, pool *mpb.Progress) (types.BlobInfo, digest.Digest, error) {
+func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, toEncrypt bool, pool *mpb.Progress) (types.BlobInfo, digest.Digest, error) {
+	// This is used to authenticate the encrypted images that are already
+	// present in the image cached. For every layer we will try to unwrap
+	// the symmetric key with the provided private keys. If we fail, we will
+	// not allow the image to be provisioned.
+	if ic.checkAuthorization {
+		if srcInfo.MediaType == manifest.DockerV2Schema2LayerGzipEncMediaType ||
+			srcInfo.MediaType == manifest.DockerV2Schema2LayerEncMediaType {
```

Comment:
I think this is a mistake on our part. We should rename `DockerV2Schema2LayerGzipEncMediaType` to use OCI instead. 

>I prefer adding an Encrypted bool field to BlobInfo and let the Manifest update it
I think this would work as well!

Kind of related to this - there is one issue that we were running into that is bugging us a little.

`copyUpdatedConfigAndManifest`(https://github.com/containers/image/blob/2178abd5f9b1b9e70c0aca05780d1defba350a53/copy/copy.go#L554) updates the manifest of the SRC and then copies it over to DST. Which is a bit tricky here because `dockerv2` doesn't support encryption, but it should not prevent encrypting a `dockerv2` image into an OCI encrypted image. I was wondering if it could be done by copying SRC to DST then applying the layer updates? What are your thoughts about this?

Reply:
I think this can be solved by doing something similar to https://github.com/containers/image/pull/563/files#diff-b3c9ca9ec48c31df0619e70595f4bd62R84-R91. While updating the manifests, we check each BlobInfo and update the mime types accordingly.

Reply:
I think the `src` vs `dst` update would still be an issue - would you think we should move the code block from https://github.com/containers/image/blob/2178abd5f9b1b9e70c0aca05780d1defba350a53/copy/copy.go#L541-L559 

down, and so do something like

```
manifest, err := ic.c.src.Manifest(...)
...
if err := ic.c.dest.PutManifest(ctx, manifest); err != nil {
		return nil, errors.Wrap(err, "Error writing manifest")
}

_, err := ic.c.dest.UpdatedImage(ctx, *ic.manifestUpdates)
if err != nil {
	return nil, errors.Wrap(err, "Error creating an updated image manifest")
}
manifest, err := ic.c.src.Manifest(...)
```